include/arch/x86/ia32/linker.ld

```diff
@@ -308,11 +308,17 @@ SECTIONS
  * zephyr_linker_sources(ROM_START ...). This typically contains the vector
  * table and debug information.
  */
+#ifndef CONFIG_LLVM_USE_BUILTIN_LLD
```

Comment:
why should those user-snippets not be included when using lld ?

Reply:
because in user-snippets, it will set local counter to 0x0, this will cause a move afterward error for those platforms whose rom start offset is not 0x0, but this could be fixed by setting CONFIG_ROM_START_OFFSET according to your suggestions, so remove this from code change.